### ISSUE TITLE
Password Protected Folders

### ISSUE THREAD

**Issue by kobergj** [owncloud/web#12039]
New menu point below `New Folder`  called `New Password Protected Folder`
* On click:
  * Open Dialog that allows setting `Foldername`  and `Password` 
  * OPTIONAL: Allow configuring the permissions on the public link
  * Create a folder in the users personal space `.PasswordProtectedFolders/projects/<space_name>/<path>/<to>/<folder>`
  * Create a public link to this folder with the given password an view permissions (or the configured permissions)
  * Create a `.url` (?) file that points to the public link
  * Either use `.ppf.url` extension or a custom one

* On clicking the `.url` file:
  * Open the `.url` file in a splash screen.

* On delete of `.url` file the public link and the hidden folder will also be deleted.
* No other actions available on the `.url` file.

**Comment by LukasHirt** [comment]
Rough estimate:

- create new app: 2
- extend extensions system to allow passing create folder actions: 5
- displaying dialog with folder name, password and link permissions: 3
- creating the hidden folder, .ppf.url file and public link: 5
- open folder in a popup: 3
- delete .ppf.url file together with the hidden folder and the public link: 2
- hide file extension and actions other than delete on the .ppf.url file: 3

**Comment by LukasHirt** [comment]
@dj4oC you showed us in the call opening the "splash screen" from some other action. Could you pls let me know what action it was? I tried to find it in the UI but has not been successful.

**Comment by dj4oC** [comment]
<img width="1912" alt="Screenshot 2024-12-10_New" src="https://github.com/user-attachments/assets/bc32bb07-d426-48e4-a412-c9e993acd3ad" />
<img width="1912" alt="Screenshot 2024-12-10_New2" src="https://github.com/user-attachments/assets/b9ff1ca1-6eb8-468a-bf80-3bf36bea277f" />
![Screenshot 2024-11-15 at 14 22 23](https://github.com/user-attachments/assets/04962c15-6e94-4da0-9325-e89b4569e4b4)
![Screenshot 2024-11-11 at 15 23 04](https://github.com/user-attachments/assets/6ac33ded-75cc-4a4d-8399-0eb55228aba4)

**Comment by dj4oC** [comment]
> @dj4oC you showed us in the call opening the "splash screen" from some other action. Could you pls let me know what action it was? I tried to find it in the UI but has not been successful.

The three dots when you have an document open as shown in the last screenshot

**Comment by LukasHirt** [comment]
> OPTIONAL: Allow configuring the permissions on the public link

I was now checking the allowed link permissions on a folder and it feels like only the "Can edit" permission really makes sense here anyway?

![Image](https://github.com/user-attachments/assets/62f1482c-6605-4f6b-a54b-8a4f041445d2)

**Comment by dj4oC** [comment]
> > OPTIONAL: Allow configuring the permissions on the public link
> 
> I was now checking the link permissions on a folder and it feels like only the "Can edit" permission really makes sense here anyway?
> 
> ![Image](https://github.com/user-attachments/assets/62f1482c-6605-4f6b-a54b-8a4f041445d2)

Can read can make sense too if the solution of a riddle has been protected. The other pupils should not change the content of the target file

**Comment by LukasHirt** [comment]
@kobergj do we consider this a core app? I.e. should it live directly in web repo or in web-extensions repo? I started working on this in web-extensions but now when I am thinking about it, it might be the wrong place? So far it's only in a PR there so it would be really easy to move https://github.com/owncloud/web-extensions/pull/130

**Comment by LukasHirt** [comment]
Agreed to move this into web repo just to simplify it and not wait for release of web.

**Comment by LukasHirt** [comment]
@kobergj all parts are implemented. Should we close this ticket or keep it open till testing/release is done?

**Comment by kobergj** [comment]
Awesome 🎆  Can be closed then 👍  But we still need a backport to `11.1` and a webbump in ocis master. Do we need a ticket for that?

**Comment by kobergj** [comment]
We can wait a bit for e2e tests I guess 👍
